source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift

```diff
@@ -0,0 +1,96 @@
+        
+        // Main loop to iterate over Array of facts
+        for fact in factArray {
+            let titleView = TitleFact()
+            let valueView = ACRFactTextField()
+            titleView.titleText.setLabel(string: fact.getTitle())
+            valueView.setLabel(string: fact.getValue())
+            
+            if let colorHex = hostConfig.getForegroundColor(style, color: .default, isSubtle: false), let textColor = ColorUtils.color(from: colorHex) {
+                titleView.titleText.labelText.textColor = textColor
+                valueView.labelText.textColor = textColor
+            }
+            
+            if !(titleView.titleText.labelText.stringValue.isEmpty) || !(valueView.labelText.stringValue.isEmpty) {
+                titleStack.addArrangedSubview(titleView.titleText)
+                valueStack.addArrangedSubview(valueView)
+            }
+        }
+        
+        // Add both elements into the Horizontal Stack
+        horizontalStack.addArrangedSubview(titleStack)
+        horizontalStack.addArrangedSubview(valueStack)
+        
+        // Add constraints once all the views are in the same view heirarchy
+        let horizontalStackViews = horizontalStack.arrangedSubviews
+        guard let titleStackagain = horizontalStackViews[0] as? NSStackView else { return NSView() }
+        guard let valueStackagain = horizontalStackViews[1] as? NSStackView else { return NSView() }
+        
+        titleStackagain.leadingAnchor.constraint(equalTo: horizontalStack.leadingAnchor).isActive = true
+        titleStackagain.topAnchor.constraint(equalTo: horizontalStack.topAnchor).isActive = true
+        titleStackagain.bottomAnchor.constraint(equalTo: horizontalStack.bottomAnchor).isActive = true
+        // Spacing between title and value in the horizontal Stack
+        titleStackagain.trailingAnchor.constraint(equalTo: valueStackagain.leadingAnchor, constant: -10).isActive = true
+        // Getting Max width from Host config if it exists
+        titleStackagain.widthAnchor.constraint(lessThanOrEqualToConstant: CGFloat(truncating: factsetConfig?.title.maxWidth ?? 150)).isActive = true
+
+        valueStackagain.trailingAnchor.constraint(equalTo: horizontalStack.trailingAnchor).isActive = true
+        valueStackagain.topAnchor.constraint(equalTo: horizontalStack.topAnchor).isActive = true
+        valueStackagain.bottomAnchor.constraint(equalTo: horizontalStack.bottomAnchor).isActive = true
+
+        // Make the height of each title and value equal
+        for (index, elem) in titleStackagain.arrangedSubviews.enumerated() {
+            guard let titleView = elem as? ACRFactTextField else { return ACRFactTextField() }
+            let valueArray = valueStackagain.arrangedSubviews
+            guard let valueView = valueArray[index] as? ACRFactTextField else { return ACRFactTextField() }
+            titleView.heightAnchor.constraint(equalTo: valueView.heightAnchor).isActive = true
```

Comment:
```suggestion
            let valueArray = valueStackagain.arrangedSubviews
            guard let titleView = elem as? ACRFactTextField, let valueView = valueArray[index] as? ACRFactTextField else { 
                 logError("// appropriate msg//")
                 continue
            }
            titleView.heightAnchor.constraint(equalTo: valueView.heightAnchor).isActive = true
```
Loop should continue on failure, shouldn't return a false value.